flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2620,8 +2621,16 @@ struct FIRToLLVMLoweringPass
         UnboxProcOpConversion, UndefOpConversion, UnreachableOpConversion,
         XArrayCoorOpConversion, XEmboxOpConversion>(context, typeConverter);
     mlir::populateStdToLLVMConversionPatterns(typeConverter, pattern);
+    mlir::populateOpenMPToLLVMConversionPatterns(typeConverter, pattern);
     mlir::ConversionTarget target{*context};
     target.addLegalDialect<mlir::LLVM::LLVMDialect>();
+    // The OpenMP dialect is legal for Operations without regions, for those
+    // which contains regions it is legal if the region contains only the
+    // LLVM dialect.
+    target.addDynamicallyLegalOp<omp::ParallelOp, omp::WsLoopOp>(
```

Comment:
We also need `MasterOp` too. 
I think the bottom line is to legalize this for every OpenMP Ops which have regions associated with it.

Reply:
Have added now. Will try to generalize (in future) this in upstream mlir conversion pass so that we can just call a function to add the legality conditions.